src/monitor_sa.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Routine that updates secrets in Spark service accounts."""
+
+import argparse
+import logging
+import os
+import sys
+from typing import Dict, Optional
+
+from lightkube.core.client import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.core_v1 import Secret, ServiceAccount
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s [%(name)s] (%(threadName)s) (%(funcName)s) %(message)s",
+)
+
+
+def read_configuration_file(file_path: str) -> Optional[Dict[str, str]]:
+    """Read spark configuration file."""
+    if not os.path.exists(file_path):
+        return None
+    with open(file_path, "r") as f:
+        lines = f.readlines()
+        confs = {}
+        for line in lines:
+            if "=" in line:
+                key_value = line.split("=")
+                confs[key_value[0]] = key_value[1]
+        return confs
+
+
+if __name__ == "__main__":
+    logger.info("Start process")
+    parser = argparse.ArgumentParser(
+        description="Handler for running a Python scripts that pushes "
+    )
+    parser.add_argument(
+        "-a",
+        "--app-name",
```

Comment:
**question** What is this argument for? I'm not sure I was able to fully understand this